oracle/src/feeds/dia_fair_price.rs

```diff
@@ -0,0 +1,120 @@
+#![allow(clippy::single_char_pattern)]
+use super::{get_http, PriceFeed};
+use crate::{config::CurrencyStore, currency::*, Error};
+use async_trait::async_trait;
+use clap::Parser;
+use reqwest::Url;
+use serde_json::Value;
+
+#[derive(Parser, Debug, Clone)]
+pub struct DiaFairPriceCli {
+    /// Fetch the exchange rate from Dia xLSD feed
+    #[clap(long)]
+    dia_fair_price_url: Option<Url>,
+}
+
+pub struct DiaFairPriceApi {
+    url: Url,
+}
+
+impl Default for DiaFairPriceApi {
+    fn default() -> Self {
+        Self {
+            url: Url::parse("https://api.diadata.org/xlsd/").unwrap(),
+        }
+    }
+}
+
+fn extract_response(value: Value, alias: &str, index: usize) -> Option<f64> {
+    let entry = value.get(index)?;
```

Comment:
This seems fragile to me - is it guaranteed that the indices don't change? I think looking up by symbol name would be more robust